modules/nrfxlib/nrf_802154/sl/platform/nrf_802154_platform_sl_lptimer_grtc_pm_utils.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "nrf_802154_platform_sl_lptimer_grtc_pm_utils.h"
+
+#include <zephyr/drivers/timer/nrf_grtc_timer.h>
+#include <zephyr/kernel.h>
+#include <zephyr/pm/policy.h>
+
+#define LPTIMER_PM_UTILS_STACK_SIZE 256
+
+static K_SEM_DEFINE(m_lptimer_pm_utils_sem, 0, 1);
+
+static struct pm_policy_event m_pm_event;
+static int64_t                m_trigger_time;
+static bool		      m_event_registered;
+
+#define SYS_TICK_PER_GRTC_CYC \
+	((uint64_t)CONFIG_SYS_CLOCK_TICKS_PER_SEC / (uint64_t)sys_clock_hw_cycles_per_sec())
+
+static int64_t grtc_ticks_to_system_ticks(uint64_t abs_grtc)
+{
+	int64_t curr_sys = k_uptime_ticks();
+	uint64_t curr_grtc = z_nrf_grtc_timer_read();
+
+	uint64_t result = (abs_grtc > curr_grtc) ? (abs_grtc - curr_grtc) : 0;
+
+	return curr_sys + result * SYS_TICK_PER_GRTC_CYC;
+}
+
+void nrf_802154_platform_sl_lptimer_pm_utils_event_update(uint64_t grtc_ticks)
+{
+	int64_t event_time = (grtc_ticks > 0) ? grtc_ticks_to_system_ticks(grtc_ticks) : 0;
```

Comment:
`grtc_ticks` is unsigned. I would recommend using
```suggestion
	int64_t event_time = (grtc_ticks != 0U) ? grtc_ticks_to_system_ticks(grtc_ticks) : 0;
```